### ISSUE TITLE
1.7.0: `inv` Causes Stack Overflow

### ISSUE THREAD

**Issue by SebastianAment** [JuliaLang/julia#43301]
Following up on [this Discourse post](https://discourse.julialang.org/t/inv-causes-stack-overflow-on-julia-1-7-0-and-mac-os/72411). According to @stevengj, this issue might be solved by the resolution [this Yggdrasil issue](https://github.com/JuliaPackaging/Yggdrasil/issues/3996).

I am running into an odd error when using Julia 1.7.0 on my 2017 Mac Book Pro (Intel i7) on Mac OS 12.01. In particular, executing
```Julia
n = 1000;
A = randn(Float32, n, n);
inv(A)
   ERROR: StackOverflowError:
```
With no further stack trace printed afterwards. 

- The same error occurs with `ComplexF32` and `ComplexF64` types, but not `Float64`.
- According to @carstenbauer, the issue is also present in 1.6.4, 1.7.0.rc-2, 1.7.0.rc-3, 1.7.0. 
- 1.6.3 and the current master branch do not exhibit the issue.

More detailed info: On my machine, inv works for all element types up to a certain size. For `Float32`, the failure size is 514 by 514 and for the complex types, it’s 258 by 258. Below these cutoffs, everything works as usual.

**Comment by ViralBShah** [comment]
Does that still happen if you set the number of blas threads to 32?

**Comment by SebastianAment** [comment]
Also, #43008 is probably related.

**Comment by giordano** [comment]
This is fixed by #43300, as shown in https://github.com/JuliaPackaging/Yggdrasil/pull/3999#issuecomment-984630053

**Comment by giordano** [comment]
> According to @carstenbauer, the issue is also present in 1.6.4, 1.7.0.rc-2, 1.7.0.rc-3, 1.7.0.

Also, I can't reproduce the issue anywhere in v1.6.4.  I have [a PR already open](https://github.com/JuliaPackaging/Yggdrasil/pull/4000) for it, but with no way to test it

**Comment by SebastianAment** [comment]
> Does that still happen if you set the number of blas threads to 32?

I had 
```Julia
julia> BLAS.get_num_threads()
2
```
It only works if I set `BLAS.set_num_threads(1)`.

**Comment by SebastianAment** [comment]
> > According to @carstenbauer, the issue is also present in 1.6.4, 1.7.0.rc-2, 1.7.0.rc-3, 1.7.0.
> 
> Also, I can't reproduce the issue anywhere in v1.6.4. I have [a PR already open](https://github.com/JuliaPackaging/Yggdrasil/pull/4000) for it, but with no way to test it

[This CI test](https://github.com/SebastianAment/LazyInverses.jl/runs/4386940669?check_suite_focus=true) of one of my packages runs 1.6-MacOS-latest (i.e. 1.6.4 to my knowledge) and fails with the same error.

**Comment by giordano** [comment]
I believe Viral was asking about _building_ OpenBLAS with 32 threads instead of 512, and the answer is always in https://github.com/JuliaPackaging/Yggdrasil/pull/3999#issuecomment-984630053

**Comment by carstenbauer** [comment]
> > According to @carstenbauer, the issue is also present in 1.6.4, 1.7.0.rc-2, 1.7.0.rc-3, 1.7.0.
> 
> Also, I can't reproduce the issue anywhere in v1.6.4. I have [a PR already open](https://github.com/JuliaPackaging/Yggdrasil/pull/4000) for it, but with no way to test it

<img width="1184" alt="Screenshot 2021-12-02 at 16 46 27" src="https://user-images.githubusercontent.com/187980/144455066-76bcda9c-784b-4270-99c2-a90aaf2af109.png">

**Comment by ViralBShah** [comment]
The bizarre thing is that it only fails with Float32, which I usually don't check manually at least. We should add all of these to the LinearAlgebra tests in Julia for size 2000.

**Comment by stevengj** [comment]
Basically this issue is a duplicate of #43008.

**Comment by giordano** [comment]
This issue should have been fixed by #43300
